tests/Toolkit/ATest.php

```diff
@@ -376,6 +401,56 @@ public function testShuffle()
 		$this->assertSame($array['bird'], $shuffled['bird']);
 	}
 
+	/**
+	 * @covers ::reduce
+	 */
+	public function testReduce()
+	{
+		$array = $this->_array();
+
+		$reduced = A::reduce($array, function ($carry, $item) {
+			return $carry . $item;
+		}, '');
+		$this->assertSame('miaowufftweet', $reduced);
+
+		$reduced = A::reduce([1, 2, 3], function ($carry, $item) {
+			return $carry + $item;
+		}, 0);
+		$this->assertSame(6, $reduced);
```

Comment:
To test that the third param is not ignored:

```suggestion
		$reduced = A::reduce([1, 2, 3], function ($carry, $item) {
			return $carry + $item;
		}, 42);
		$this->assertSame(48, $reduced);
```

Reply:
That is a great idea